components/Onboarding.js

```diff
@@ -115,26 +113,10 @@ const styles = StyleSheet.create({
 
 class Onboarding extends PureComponent {
   componentDidUpdate(prevProps) {
-    const {
-      numLocations,
-      completed: { hasAddedLocation, hasAcceptedInvite, hasViewedPin },
-      step,
-      setCompleted,
-      setStep,
-    } = this.props;
-
-    if (((numLocations > prevProps.numLocations && !hasAddedLocation) || hasAddedLocation) && step === STEPS.addLocationAction) {
-      setStep(STEPS.viewPinCallout);
-      setCompleted(COMPLETED_KEYS.hasAddedLocation);
-    }
-
-    if (hasViewedPin && (step === STEPS.viewPinCallout || step === STEPS.viewPinCalloutAction)) {
-      setStep(STEPS.pinCalloutDescription);
-    }
+    const { step } = this.props;
 
-    if (this.props.regionKey && (step === STEPS.invitations1 || step === STEPS.invitationsAction) && !hasAcceptedInvite) {
-      setCompleted(COMPLETED_KEYS.hasAcceptedInvite);
-      setStep(STEPS.invitationAccepted);
+    if (step < STEPS.end && STEPS[ORDERED_STEPS[step]].actionLogic) {
```

Comment:
Using `componentDidUpdate` may be problematic. React is trying to phase this out.

Reply:
They are? Do they suggest a lifecycle method to use instead? (I couldn't find anything myself)

Reply:
I would highly recommend reading https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html